earthmover/operations/column.py

```diff
@@ -82,36 +82,42 @@ def execute(self, data: 'DataFrame', **kwargs) -> 'DataFrame':
         super().execute(data, **kwargs)
 
         for col, val in self.columns_dict.items():
-
-            # Apply the value as a static string if not obviously Jinja.
-            if not util.contains_jinja(val):
-                data[col] = val
-
+            # Apply value to all columns
```

Comment:
I'd like to learn more about the context behind this PR. Do we want to allow wildcard columns with prefixes or suffixes, or just "select-all"?

Reply:
The Monday board states it could be useful to remove white-spaces in all values of the df. Its currently "select-all," but I am looking for a rework to use REGEX. You can see the Monday Board comment below if you would like more info:



This idea came from Ryan's comment about needing to [trim](https://jinja.palletsprojects.com/en/3.1.x/templates/#jinja-filters.trim) whitespace off all values in a dataframe. I propose we could extend the `modify_columns` operation to allow the following:

      - operation: modify_columns
        columns:
          *: "{%raw%}{{value|trim}}{%endraw%}"

(This would make "*" a reserved column name, but I think that's fine.)

Under the hood, earthmover can look for the asterisk and just loop over columns in the dataframe and apply the Jinja transformation to all of them.

Reply:
This makes sense. If we allow regex for dynamic column selection, we may need a way to distinguish regex from non-regex columns. There's nothing stopping columns in Pandas from containing special regex characters like parentheses and asterisks.

Reply:
Adding my 2 cents: [here](https://github.com/edanalytics/earthmover/blob/3abb434801820c14e33e25c25e26f533d69f289a/earthmover/earthmover.py#L327-L329), we currently use asterisks for node selection (e.g., `node_*`). The actual selection logic is implemented [here](https://github.com/edanalytics/earthmover/blob/3abb434801820c14e33e25c25e26f533d69f289a/earthmover/graph.py#L87-L91). I propose the same functionality - it could be abstracted to its own reusable function, perhaps `util.string_select(selector_string, list_of_strings)` - for `modify_columns` (and, in the future, perhaps for `drop_columns`, `keep_columns`, `map_values`, and possibly `date_format`) - so all of the following would be legal:
```yaml
# given a $sources.X with columns `a_1`, `a_2`, `b_1`, `b_2`, `a_1_b_2`, `b_1_a_2`
transformations:
  my_test:
    source: $sources.X
    operations:
      - operation: modify_columns
        columns:
        a_*: "{%raw%}{{value|trim}}{%endraw%}" # modifies `a_1`, `a_2`, `a_1_b_2`
        *_1: "{%raw%}{{value|trim}}{%endraw%}" # modifies `a_1`, `b_1`
        *_1_*: "{%raw%}{{value|trim}}{%endraw%}" # modifies `a_1_b_2`, `b_1_a_2`
        *: "{%raw%}{{value|trim}}{%endraw%}" # modifies `a_1`, `a_2`, `b_1`, `b_2`, `a_1_b_2`, `b_1_a_2` (all columns)
```

You're right, @jayckaiser , that technically nothing prevents column names in Pandas from containing an asterisk, however in practice I've never seen source data with such a column. I think as long as we document this limitation it should be fine for earthmover.

Thoughts?